Clarify license types and IT type (#4766)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/manage-access/cloud-seats-and-users.md

- **Developer** &mdash; Granted access to the Deployment and [Development](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud) functionality in dbt Cloud.
- **Read-Only** &mdash; Intended to view the [artifacts](/docs/deploy/artifacts) created in a dbt Cloud account. Read-Only users can receive job notifications but not configure them.

> - **IT** &mdash; Can manage users, groups, and licenses, among other permissions. IT users can receive job notifications but not configure them. Available on Enterprise and Team plans only.  In Enterprise plans, the IT license type grants access equivalent to the ['Security admin' role](/docs/cloud/manage-access/enterprise-permissions#account-permissions-for-account-roles). 

Reviewer: ```suggestion
- **IT** &mdash; Can manage users, groups, and licenses, among other permissions. IT users can receive job notifications but not configure them. Available on Enterprise and Team plans only.  In Enterprise plans, the IT license type grants access equivalent to the ['Security admin' and 'Billing admin' roles](/docs/cloud/manage-access/enterprise-permissions#account-permissions-for-account-roles). 
```

> - **IT** &mdash; Can manage users, groups, and licenses, among other permissions. IT users can receive job notifications but not configure them. Available on Enterprise and Team plans only. On Enterprise plans - the IT license type provides the same access as the ["Security admin" permission set / role](https://docs.getdbt.com/docs/cloud/manage-access/enterprise-permissions#account-permissions-for-account-roles).

Reviewer: @jeremyyeo should we also add 'billing admin' per [this doc](https://docs-getdbt-g45daipnq-dbt-labs.vercel.app/docs/cloud/manage-access/about-user-access#license-based-access-control). 

I've asked aaron here: https://dbtlabs.atlassian.net/browse/ENTERPRISE-1212

```suggestion
- **IT** &mdash; Can manage users, groups, and licenses, among other permissions. IT users can receive job notifications but not configure them. Available on Enterprise and Team plans only.  In Enterprise plans, the IT license type grants access equivalent to the ['Security admin' and 'Billing admin' roles](/docs/cloud/manage-access/enterprise-permissions#account-permissions-for-account-roles). 
```
